### PR TITLE
Ovirt vnic profile fix

### PR DESCRIPTION
**Pull request by mnecas**
##### SUMMARY

update vnic profile 

BZ 1597537
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

ovirt
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/ovirt/ovirt_vnic_profile.py [hunk 1]
```diff
@@ -149,45 +149,62 @@ class EntityVnicPorfileModule(BaseModule):
...
-    def __get_network_filter_id(self):
+    def _get_network_filter_id(self):
         nf_service = self._connection.system_service().network_filters_service()
         return get_id_by_name(nf_service, self.param('network_filter')) if self.param('network_filter') else None
 
+    def _get_network_filter(self):
+        network_filter = None
+        if self.param('network_filter'):
+            network_filter = otypes.NetworkFilter(id=self._get_network_filter_id())
+        elif self.param('network_filter') == '' or self.param('pass_through') == 'enabled':
```
Comment (mnecas) [inline lib/ansible/modules/cloud/ovirt/ovirt_vnic_profile.py:175]: @machacekondra 
should user remove the network_filter manually with the pass_through 'enabled' or keep it automatically ?

File: lib/ansible/modules/cloud/ovirt/ovirt_vnic_profile.py [hunk 2]
```diff
@@ -149,45 +149,62 @@ class EntityVnicPorfileModule(BaseModule):
...
+            network_filter = otypes.NetworkFilter()
+        return network_filter
+
+    def _get_qos(self):
+        qos = None
+        if self.param('qos'):
+            qos = otypes.Qos(id=self._get_qos_id())
+        elif self.param('qos') == '' or self.param('pass_through') == 'enabled':
```
Comment (mnecas) [inline lib/ansible/modules/cloud/ovirt/ovirt_vnic_profile.py:183]: same here

File: lib/ansible/modules/cloud/ovirt/ovirt_vnic_profile.py [hunk 3]
```diff
@@ -149,45 +149,62 @@ class EntityVnicPorfileModule(BaseModule):
...
+            qos = otypes.Qos()
+        return qos
+
+    def _get_port_mirroring(self):
+        if self.param('pass_through') == 'enabled':
```
Comment (mnecas) [inline lib/ansible/modules/cloud/ovirt/ovirt_vnic_profile.py:188]: and here

File: lib/ansible/modules/cloud/ovirt/ovirt_vnic_profile.py [hunk 4]
```diff
@@ -149,45 +149,62 @@ class EntityVnicPorfileModule(BaseModule):
...
+            return False
+        return self.param('pass_through')
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_vnic_profile.py]: s/pass_through/port_mirroring

File: lib/ansible/modules/cloud/ovirt/ovirt_vnic_profile.py [hunk 5]
```diff
@@ -243,11 +260,27 @@ def main():
...
+            # When vNIC already exist update it, when not create it.
+            if vnic.name == module.params['name'] and network_id == vnic.network.id:
+                entity = vnic
+                force_create = False
+                break
+
         if state == 'present':
-            ret = entitynics_module.create()
+            if entity is not None and entity.pass_through == 'disabled' and module.params.get('pass_through') == 'enabled':
+                module.params['port_mirroring'] = False
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_vnic_profile.py]: Seems unneeded

File: lib/ansible/modules/cloud/ovirt/ovirt_vnic_profile.py (hunk 1 above)
Comment (machacekondra) [reply lib/ansible/modules/cloud/ovirt/ovirt_vnic_profile.py:175]: Please document it, how it behaves.

**Review by machacekondra — APPROVED** [review]
shipit

**Pull request merged**